telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
+
+def run_benchmarks(num_urls):
+    """Runs the page_cycler benchmark for each url
+
+    Dumps data /temp/benchmark_results.db
+    :param num_urls: int of the number of urls to run through
+    """
+    path.append('/home/jamshed/src/tools/perf/')
+    cmd = ('sudo /home/jamshed/src/tools/perf/run_benchmark '
+           'page_cycler.PageCyclerUrl{0}')
+    for i in range(num_urls):
+        p = Popen(cmd.format(i), shell=True)
+        p.wait()  # We can only load 1 url on Chrome at a time
+        # Also run the modified version
+        p = Popen(cmd.format('999999' + str(i)), shell=True)
+        p.wait()  # We can only load 1 url on Chrome at a time
+
+def aggregate_benchmark_data():
+    """Merges data from /temp/benchmark_results.db with /temp/aggregate.db
+
+    Note: temp/aggregate.db looks like {'http://jamnoise.com':
+                                           {'cold_times':
+                                               {'trial1': [data],...}}}
+    """
+    benchmark_results = '/data/benchmark_results.db'
+    benchmark_result_path = \
+            '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db'
+    aggregate_path = \
+            '/home/jamshed/page_load_time/telemetry/temp/aggregate.db'
+
+    # Get current benchmark data and aggregate data
+    if not os.path.isfile(benchmark_result_path):
+        # Ensure path exists
+        tmp = {}
+        pickle.dump(tmp, open(benchmark_result_path, 'wb'))
+    if not os.path.isfile(aggregate_path):
+        # Ensure path exists
+        tmp = {}
+        pickle.dump(tmp, open(aggregate_path, 'wb'))
+    try:
+        curr_benchmark_data = pickle.load(open(benchmark_result_path, 'rb'))
+        curr_aggregate_data = pickle.load(open(aggregate_path, 'rb'))
+    except IOError:
+        # Raise exception or rewrite benchmark_results.db?
+        raise IOError('Failed to read benchmark or aggregate data path')
+
+    # Get url mappings
+    url_mappings = {}
+    try:
+        url_mappings = pickle.load(open('data/url_mapping.db', 'rb'))
+    except IOError:
+        raise IOError('Failed to read url mapping from data/url_mapping.db')
+    if url_mappings == {}:
+        raise KeyError('url_mappings is empty')
+
+    # Enforce benchmark data only has one trial
+    if len(curr_benchmark_data.keys()) != 1:
+        raise KeyError('Current benchmark data contains > 1 trial')
+
+    # Merge data, use url mapping
+    trial_number = curr_benchmark_data.keys()[0]
+    trial_name = 'trial{0}'.format(trial_number)
+    print "Trial: " + str(trial_number)
+
+    url_regex = 'url(\d+)'
+    for urlId in curr_benchmark_data[trial_number]:
+        measurement = 'cold_times'
+        if '999999' in str(urlId):
+            measurement = 'modified_cold_times'
+
+        curr_id = int(findall(url_regex, urlId)[0])
+        url_name = url_mappings[curr_id]
+        cold_time_data = \
+        eval(str(curr_benchmark_data[trial_number][urlId]['cold_times']))
```

Comment:
I'm not sure I understand why we don't store an array to begin with? Why do we store the string `'[132.134, 100, 12.632]'`, rather than the array `[132.134, 100, 12.632]`?